codec-registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala

```diff
@@ -176,6 +160,69 @@ class SerializerCheckCompilerPluginComponent(
             None
         }
       }
+
+      private def collectMissingClassNames(fullyQualifiedClassNames: List[String]): List[String] = {
+        val sourceCodeDir = File(options.sourceCodeDirectoryToCheck)
+        val sourceCodeFilesAsStrings =
+          (for (file <- sourceCodeDir.collectChildren(_.name.endsWith(".scala"))) yield file.contentAsString).toList
+
+        def typeIsDefinedInScalaFiles(fqcn: String): Boolean = {
+          val indexOfLastDotInFQCN = fqcn.lastIndexOf('.')
+          val packageName = fqcn.substring(0, indexOfLastDotInFQCN)
+          val typeName = fqcn.substring(indexOfLastDotInFQCN + 1)
+          sourceCodeFilesAsStrings.exists(fileAsString => {
+            fileAsString.startsWith(s"package $packageName") &&
+            (fileAsString.contains(s"class $typeName") || fileAsString.contains(s"trait $typeName") || fileAsString
+              .contains(s"object $typeName"))
+          })
+        }
+
+        fullyQualifiedClassNames.filter(typeIsDefinedInScalaFiles)
+      }
+
+      private def interactWithTheCacheFile(
```

Comment:
Actually, given that this method is only ever called with a specific `mode` known upfront (and not a variable one)... it'll be better to split this method into two independent methods (esp. that the 2 `match` cases are completely separate)

Reply:
done